lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1325,9 +1325,22 @@ SwiftLanguageRuntimeImpl::BindGenericTypeParameters(StackFrame &stack_frame,
   }
 
   swift::Demangle::Demangler dem;
-  swift::Demangle::NodePointer canonical =
-      TypeSystemSwiftTypeRef::GetCanonicalDemangleTree(
-          ts.GetModule(), dem, mangled_name.GetStringRef());
+  swift::Demangle::NodePointer canonical = TypeSystemSwiftTypeRef::Transform(
+      dem, dem.demangleSymbol(mangled_name.GetStringRef()),
+      [](swift::Demangle::NodePointer node) {
+        if (node && node->getKind() == Node::Kind::DynamicSelf) {
```

Comment:
1. `node` is always non-null
2. maybe make this this be an early exit condition like those below

Reply:
Good point. It's unfortunate that there is no `NodeReference` typedef.